src/join/composer.jl

```diff
@@ -207,21 +220,49 @@ function compose_joined_table(joiner::DataFrameJoiner, kind::Symbol, makeunique:
 
     @assert col_idx == ncol(joiner.dfl_on) + 1
 
-    for col in eachcol(dfl_noon)
-        cols_i = left_idxs[col_idx]
-        cols[cols_i] = _similar_left(col, target_nrow)
-        copyto!(cols[cols_i], view(col, left_ixs))
-        copyto!(cols[cols_i], lil + 1, view(col, leftonly_ixs), 1, loil)
-        col_idx += 1
-    end
-
-    @assert col_idx == ncol(joiner.dfl) + 1
-
-    for col in eachcol(dfr_noon)
-        cols[col_idx] = _similar_right(col, target_nrow)
-        copyto!(cols[col_idx], view(col, right_ixs))
-        copyto!(cols[col_idx], lil + loil + 1, view(col, rightonly_ixs), 1, roil)
-        col_idx += 1
+    @static if VERSION >= v"1.4"
+        if Threads.nthreads() > 1 && target_nrow >= 1_000_000 && length(cols) > col_idx
+            @sync begin
+                for col in eachcol(dfl_noon)
+                    cols_i = left_idxs[col_idx]
+                    Threads.@spawn _noon_compose_helper(cols, _similar_left, cols_i,
+                                                        col, target_nrow, left_ixs, lil + 1, leftonly_ixs, loil)
+                    col_idx += 1
+                end
+                @assert col_idx == ncol(joiner.dfl) + 1
+                for col in eachcol(dfr_noon)
+                    let cols_i = col_idx
```

Comment:
Why do you use `let` here but not above?

Reply:
You are right - `let` is not needed, as just writing `cols_i = col_idx` will ensure we get a new binding for `cols_i`. I will change this.